src/wallet/external_signer_scriptpubkeyman.cpp

```diff
@@ -94,11 +94,14 @@ std::optional<PSBTError> ExternalSignerScriptPubKeyMan::FillPSBT(PartiallySigned
     if (complete) return {};
 
     auto signer{GetExternalSigner()};
-    if (!signer) throw std::runtime_error(util::ErrorString(signer).original);
+    if (!signer) {
+        LogWarning("%s", util::ErrorString(signer).original);
+        return PSBTError::EXTERNAL_SIGNER_NOT_FOUND;
```

Comment:
Perhaps we could have a test for it since it doesn't throw a runtime error anymore?

Reply:
Added a functional test.

Note that this test would pass on the original code as well, with just a slightly different error:

```diff
diff --git a/test/functional/wallet_signer.py b/test/functional/wallet_signer.py
index 58705a23a3..07f18124c9 100755
--- a/test/functional/wallet_signer.py
+++ b/test/functional/wallet_signer.py
@@ -259,7 +259,7 @@ class WalletSignerTest(BitcoinTestFramework):

         # Try to spend
         dest = hww.getrawchangeaddress()
-        assert_raises_rpc_error(-25, "External signer not found", hww.send, outputs=[{dest:0.5}])
+        assert_raises_rpc_error(-1, "GetExternalSigner: No external signers found", hww.send, outputs=[{dest:0.5}])
```

A unit test might be able to cover `PSBTError::EXTERNAL_SIGNER_NOT_FOUND` directly.